src/components/gridList/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useCallback, useMemo} from 'react';
+import {FlatList, FlatListProps} from 'react-native';
+import {Spacings} from 'style';
+import {useOrientation} from 'hooks';
+import {Constants} from '../../commons/new';
+import View from '../view';
+
+const DEFAULT_NUM_COLUMNS = 3;
+const DEFAULT_ITEM_SPACINGS = Spacings.s4;
+
+export interface GridListProps<T> extends FlatListProps<T> {
+  /**
+   * Allow a responsive item width to the maximum item width
+   */
+  maxItemWidth?: number;
+  /**
+   * Number of items to show in a row (ignored when passing maxItemWidth)
+   */
+  numColumns?: number;
+  /**
+   * Spacing between each item
+   */
+  itemSpacing?: number;
+  /**
+   * List padding (used for item size calculation)
+   */
+  listPadding?: number;
```

Comment:
The `listPadding` is not acting as expected, for example change the screen to use:
```
numColumns={2}
// maxItemWidth={140}
itemSpacing={Spacings.s10}
listPadding={Spacings.s2}
```
There is no padding on the right side.

Or:
```
numColumns={2}
// maxItemWidth={140}
itemSpacing={Spacings.s10}
// listPadding={Spacings.s2}
```
It looks like the right item is a little out of the screen

Reply:
Well, technically the `listPadding` prop goes together with setting the list `contentContainerStyle` with padding. 
But I see your point, I think I'll inject the padding in the list myself so the user won't have to worry about it